nuxeo-features/nuxeo-platform-comment/nuxeo-platform-comment-api/src/main/java/org/nuxeo/ecm/platform/comment/api/AnnotationService.java

```diff
@@ -44,37 +46,40 @@
      *
      * @param session the core session
      * @param annotationId the annotation document model id
+     * @throws NuxeoException if no annotation was found with the given id
      * @return the annotation, or {@code null} if the annotation with the given id was not found
      */
-    Annotation getAnnotation(CoreSession session, String annotationId);
+    Annotation getAnnotation(CoreSession session, String annotationId) throws NuxeoException;
 
     /**
      * Gets all annotations for a document.
      *
      * @param session the core session
      * @param documentId the annotated document id
      * @param xpath the annotated blob xpath in the document
+     * @throws NuxeoException if no annotation was found with the given id
      * @return the list of annotations, or an empty list if no annotation is found
      */
-    List<Annotation> getAnnotations(CoreSession session, String documentId, String xpath);
+    List<Annotation> getAnnotations(CoreSession session, String documentId, String xpath) throws NuxeoException;
 
     /**
      * Updates an annotation for a document.
      *
      * @param session the core session
      * @param annotationId the annotation document model id
      * @param annotation the annotation containing the modifications
+     * @throws NuxeoException if no annotation was found with the given id
      */
-    void updateAnnotation(CoreSession session, String annotationId, Annotation annotation);
+    void updateAnnotation(CoreSession session, String annotationId, Annotation annotation) throws NuxeoException;
 
     /**
      * Deletes an annotation for a document.
      *
      * @param session the core session
      * @param annotationId the annotation document model id
-     * @throws IllegalArgumentException if no annotation was found with the given id
+     * @throws NuxeoException if no annotation was found with the given id
```

Comment:
It shouldn't throw at all if nothing was deleted.

Reply:
Mixed feelings about that... Assuming that annotations are similar to comments, we want to return a 404 when we try to delete a comment for a non existing document.